aeon/dj_pipeline/spike_sorting.py

```diff
@@ -229,6 +228,23 @@ def make_compute(self, key, ephys_files, dir_types, electrodes_df,
         si_recording = ephys_preproc(si_recording)
         si_recording.dump_to_pickle(file_path=recording_file, relative_to=output_dir)
 
+        # write binary into recording.dat
+        job_kwargs = {"n_jobs": 0.8, "chunk_duration": "2s"}
+        binary_file_path = recording_file.parent / "recording.dat"
+        if binary_file_path.exists():
+            load_and_verify_binary_file(
+                binary_file_path=binary_file_path,
+                se_recording_obj=si_recording)
+            logger.info(f"{binary_file_path} already exists. Skipping writing binary file.")
+        else:
+            logger.info(f"Writing binary recording to {binary_file_path}...")
+            si.core.write_binary_recording(
+                    recording=si_recording,
+                    file_paths=[binary_file_path],
+                    dtype="int16",
+                    **sorters.basesorter.get_job_kwargs(job_kwargs, True),
+                )
```

Comment:
We call `load_and_verify_binary_file` twice:
once in `PreProcessing.make_compute` (but only if the bin file exists):
https://github.com/SainsburyWellcomeCentre/aeon_mecha/blob/66dea030c45658272f5c035e9e6a3bf1e090bfc0/aeon/dj_pipeline/spike_sorting.py#L234-L238) 
and later on in `SpikeSorting.make_compute`:
https://github.com/SainsburyWellcomeCentre/aeon_mecha/blob/66dea030c45658272f5c035e9e6a3bf1e090bfc0/aeon/dj_pipeline/spike_sorting.py#L315-L318

We could instead call `load_and_verify_binary_file` once in SpikeSorting
```suggestion
        if binary_file_path.exists():
            logger.info(f"{binary_file_path} already exists. Skipping writing binary file.")
        else:
            logger.info(f"Writing binary recording to {binary_file_path}...")
            si.core.write_binary_recording(
                    recording=si_recording,
                    file_paths=[binary_file_path],
                    dtype="int16",
                    **sorters.basesorter.get_job_kwargs(job_kwargs, True),
                )
```

Reply:
It's an additional safety check to make sure the existing binary file is indeed the expected file (matches the original recording object)